transfers/amclient.py

```diff
@@ -0,0 +1,518 @@
+
+    def transferables(self, b64decode=False):
+        """Return all transferable entities in the Storage Service.
+        GET location/<TS_LOC_UUID>/browse/::
+
+            $ ./amclient.py transferables \
+                --ss-user=test \
+                --ss-api-key=7558e7485cf8f20aadbd95f6add8b429ba11cd2b \
+                --transfer-source=7ea1eb0e-5f4e-42e0-836d-c9b4ab5692e1 \
+                --transfer-path=vagrant/archivematica-sampledata
+
+        """
+        url = '{}/api/v2/location/{}/browse/'.format(
+            self.ss_url, self.ts_location_uuid)
+        params = self._ss_auth()
+        if self.ts_path_prefix:
+            params['path'] = base64.b64encode(self.ts_path_prefix)
+        result = self._call_url_json(url, params)
+        if b64decode:
+            return self.b64decode_ts_location_browse(result)
+        return result
+
+    def b64decode_ts_location_browse(self, result):
+        """Base64-decode the results of a call to SS GET
+        /location/UUID/browse/.
+        """
+        def dec(thing):
+            return base64.b64decode(getattr(thing, 'encode')('utf8'))
+        try:
+            result['directories'] = [dec(d) for d in result['directories']]
+            result['entries'] = [dec(e) for e in result['entries']]
+            result['properties'] = dict((dec(k), v) for k, v in
+                                        result['properties'].items())
+            return result
+        except Exception:
+            return result
+
+    def search_package(self, params):
+        """SS GET /search/package?<GET_PARAMS>."""
+        return self._call_url_json(
+            '{}/api/v2/search/package/'.format(self.ss_url), params)
+
+    def stdout(self, stuff):
+        """Print to stdout, either as JSON or pretty-printed Python."""
+        if self.output_mode == 'json':
+            print(json.dumps(stuff))
+        else:
+            pprint.pprint(stuff)
+
+    def aips(self):
+        return self.get_all_packages('AIP')
+
+    def dips(self):
+        return self.get_all_packages('DIP')
+
+    def get_all_packages(self, ptype='AIP', packages=None, next=None):
+        """Get all packages (AIPs or DIPs) in the Storage Service, following
+        the pagination trail if necessary.
+
+        WARNING: this method requires a version of the Storage Service with a
+        search API, cf. dev/issue-8895-search-api/.
+        """
+        if not packages:
+            packages = []
+        if next:
+            r = requests.get(next).json()
+        else:
+            r = self.search_package({'package_type': ptype})
+        packages = packages + r['results']
+        if r['next']:
+            packages = self.get_all_packages(ptype, packages, r['next'])
+        return packages
+
+    def aip2dips(self):
+        """Get all DIPS created from AIP with UUID ``self.aip_uuid``."""
+        dips = self.get_all_dips()
+        return [d for d in dips if
```

Comment:
Can this be done by only getting DIPs that related to an AIP UUID using tastypie's filters? https://github.com/artefactual/archivematica-storage-service/blob/stable/0.9.x/storage_service/locations/api/resources.py#L434 should be relevant.